x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -389,9 +400,9 @@ export class TaskRunner<
           executionId: this.executionId,
           services: {
             savedObjectsClient,
+            searchSourceUtils,
             uiSettingsClient: this.context.uiSettings.asScopedToClient(savedObjectsClient),
             scopedClusterClient: wrappedScopedClusterClient.client(),
-            searchSourceClient: this.context.data.search.searchSource.asScoped(fakeRequest),
```

Comment:
**question:** Could we pass a `wrappedSearchSourceClient` that has the same interface as `searchSourceClient` but with the `fetch` function wrapped? We've tried this approach before with the Elasticsearch client but found a lot of places like to pass an instance of `esClient` to their libraries and it was easier for us to inject a new "search" into the client rather than a separate `search` function. 

I fear if we expose `searchSourceUtils`, it allows rule types to choose if they want to choose a `fetch` that is cancellable or not, when it should always be cancellable.

Thoughts?

Reply:
If think we need searchSourceUtils because we need `searchSourceClient.create`, right @Dmitriynj ?

Reply:
I've suggested solution in the latest commits. To pass `searchSourceClient` we need `searchSourceClient.create`, which should create wrapped `searchSource` instance, that is true. Also we there is `searchSourceClient.createEmpty` method.

Apart from that we need to consider further child instances creation of `searchSource`. To be exact: `searchSource.createChild`, `searchSource.createCopy`, `searchSource.create` should return wrapped instances of `searchSource`.

All of that added to `wrap_search_source_client.ts`.

Reply:
@mikecote  this is a lot of wrapping we need to do, do you think this is mandatory?

Reply:
>@mikecote this is a lot of wrapping we need to do, do you think this is mandatory?

I would say it's not mandatory but highly recommended. It guarantees every fetch / search is cancelled when an alerting rule is cancelled / timed out as the adoption of searchSourceClient grows. I took a quick look at `wrap_search_source_client.ts` and it doesn't seem too bad, we might be able to further simplify it but I'll take a second look first.

Reply:
Is there a way the `SearchSource` class within the data plugin could be enhanced to take an `abortController` as an argument and use it for the current, child, and copied instances? I'm wondering if there's a way this could be done without needing to wrap at all 🤔

Reply:
I think this is a good question for @elastic/kibana-app-services, can you help in this case?

Reply:
> Is there a way the search service or search source within the data plugin could take an AbortController and re-use it everywhere for a given alerting rule run?

Are you suggesting something like this? 

```
const abortController = new AbortController();
const searchSource = new SearchSource({...params, abortSignal: abortController.signal})

searchSource.fetch();

searchSource.fetch();

searchSource.fetch();


abortController.abort(); // cancels everything and throws in case searchSource.fetch() is called again
```

will something like this help?

Reply:
One point we should consider is if `searchSource` will be created in task runner, we need to fill it with params similar in [es_query alert executor](https://github.com/elastic/kibana/pull/130077/files#diff-8ccf6feabc5b3b61c8ac9e4ab8036ef9379b8bc22e7a8887645dd7a7eaf45c20R28): 
```
await searchSourceClient.create(params.searchConfiguration)
```

Reply:
Thanks, 
I think that would be nice to have on search source , but I think if currently implemented proxy approach doesn't have any downsides (except boilerplate code) then I don't think we should block it. 

Currently implementation also has logging / telemetry additions, right? Which  I assume we would lose if we implemented that abortable SearchSource

Reply:
@kertal absolutely, not a problem. I will review the current wrapping approach 👍